Update Hyper and OpenSSL

nox opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15868)
<!-- Reviewable:end -->

avadacatavra committed:
Update Hyper and OpenSSL

highfive commented:
Heads up! This PR modifies the following files:
 * @asajeffrey: components/webdriver_server/lib.rs, components/webdriver_server/Cargo.toml
 * @jgraham: components/webdriver_server/lib.rs, components/webdriver_server/Cargo.toml
 * @fitzgen: components/devtools_traits/Cargo.toml, components/devtools_traits/Cargo.toml, components/devtools/Cargo.toml, components/script/dom/document.rs, components/script/webdriver_handlers.rs, components/script/Cargo.toml, components/script/dom/websocket.rs, components/script_traits/webdriver_msg.rs, components/script_traits/webdriver_msg.rs, components/script_traits/Cargo.toml, components/script_traits/Cargo.toml
 * @kichjang: components/net/lib.rs, components/script/dom/document.rs, components/net/http_loader.rs, components/net/connector.rs, components/net_traits/lib.rs, components/net_traits/lib.rs, components/script/webdriver_handlers.rs, components/script/Cargo.toml, components/net/cookie_storage.rs, components/net_traits/Cargo.toml, components/net_traits/Cargo.toml, components/net/resource_thread.rs, components/net/Cargo.toml, components/net/cookie.rs, components/net/subresource_integrity.rs, components/script/dom/websocket.rs, components/script_traits/webdriver_msg.rs, components/script_traits/webdriver_msg.rs, components/script_traits/Cargo.toml, components/script_traits/Cargo.toml

nox commented:
@bors-servo r+ p=17

bors-servo commented:
:pushpin: Commit 497a7b2 has been approved by `nox`

<!-- @bors-servo r=nox 497a7b2dab4e9c262e67e32ab2da6048b6e3a7f4 -->

bors-servo commented:
:hourglass: Testing commit 497a7b2 with merge 788109e...

bors-servo commented:
:broken_heart: Test failed - [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1174)

nox commented:
```
Build failed, waiting for other jobs to finish...
error: failed to run custom build command for `openssl-sys v0.9.7`
process didn't exit successfully: `c:\buildbot\slave\windows-msvc-dev\build\target\debug\build\openssl-sys-ab551cd13dbf3163\build-script-build` (exit code: 101)
--- stdout
cargo:rustc-link-search=native=C:\Users\Administrator\.servo\msvc-dependencies\openssl\1.0.1t-vs2015\lib64
cargo:include=C:\Users\Administrator\.servo\msvc-dependencies\openssl\1.0.1t-vs2015\include
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
CC_x86_64_pc_windows_msvc = None
HOST_CC = None
CC = None
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
PROFILE = Some("debug")
running: "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "C:\\Users\\Administrator\\.servo\\msvc-dependencies\\openssl\\1.0.1t-vs2015\\include" "/E" "c:\\buildbot\\slave\\windows-msvc-dev\\build\\target\\debug\\build\\openssl-sys-b989cad2fc000929\\out\\expando.c"
cargo:warning=expando.c
ExitStatus(ExitStatus(0))
cargo:rustc-cfg=osslconf="OPENSSL_NO_KRB5"
cargo:rustc-cfg=osslconf="OPENSSL_NO_RFC3779"
cargo:conf=OPENSSL_NO_KRB5,OPENSSL_NO_RFC3779
cargo:rustc-cfg=ossl101
cargo:version=101

--- stderr
thread 'main' panicked at 'OpenSSL libdir at `C:\Users\Administrator\.servo\msvc-dependencies\openssl\1.0.1t-vs2015\lib64` does not contain the required files to either statically or dynamically link OpenSSL', C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\openssl-sys-0.9.7\build.rs:355
stack backtrace:
   0:      0x7f6083964c7 - std::panicking::default_hook::{{closure}}
                        at C:\projects\rust\src\libstd\panicking.rs:356
   1:      0x7f6083959f4 - std::panicking::default_hook
                        at C:\projects\rust\src\libstd\panicking.rs:367
   2:      0x7f608399311 - std::panicking::rust_panic_with_hook
                        at C:\projects\rust\src\libstd\panicking.rs:545
   3:      0x7f6083991c8 - std::panicking::begin_panic<collections::string::String>
                        at C:\projects\rust\src\libstd\panicking.rs:507
   4:      0x7f6083990e4 - std::panicking::begin_panic_fmt
                        at C:\projects\rust\src\libstd\panicking.rs:491
   5:      0x7f608316f68 - build_script_build::determine_mode
                        at C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\openssl-sys-0.9.7\build.rs:355
   6:      0x7f60831468f - build_script_build::main
                        at C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\openssl-sys-0.9.7\build.rs:76
   7:      0x7f60839bbc1 - panic_unwind::__rust_maybe_catch_panic
                        at C:\projects\rust\src\libpanic_unwind\lib.rs:98
   8:      0x7f60839991a - std::rt::lang_start
                        at C:\projects\rust\src\libstd\rt.rs:51
   9:      0x7f60831725b - main
  10:      0x7f6083a05d8 - __scrt_common_main_seh
                        at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:253
  11:      0x7fd5dab1841 - BaseThreadInitThunk
```

Cc @sfackler @retep998 @avadacatavra

sfackler commented:
What is in C:\Users\Administrator\.servo\msvc-dependencies\openssl\1.0.1t-vs2015\lib64?

Ms2ger commented on tests/unit/net/fetch.rs:
```diff
@@ -18,10 +18,11 @@ use hyper::header::{Encoding, Location, Pragma, Quality, QualityItem, SetCookie,
 use hyper::method::Method;
 use hyper::mime::{Mime, SubLevel, TopLevel};
-use hyper::net::Openssl;
+//use hyper::net::Openssl;
```
Remove pls

Ms2ger commented on components/net/resource_thread.rs:
```diff
@@ -307,9 +310,11 @@ impl CoreResourceManager {
-                          resource_group: &ResourceGroup) {
-        if let Some(cookie) = cookie::Cookie::new_wrapped(cookie, &request, source) {
+    fn set_cookie_for_url(&mut self, request: &ServoUrl,
+                           cookie: cookie_rs::Cookie<'static>,
```
Indent

Ms2ger commented on components/net/http_loader.rs:
```diff
@@ -126,42 +125,62 @@ struct NetworkHttpRequestFactory {
                                                       &*self.connector);
 
         if let Err(HttpError::Ssl(ref error)) = connection {
             let error: &(Error + Send + 'static) = &**error;
```
Still necessary?

Ms2ger commented on components/net/http_loader.rs:
```diff
@@ -126,42 +125,62 @@ struct NetworkHttpRequestFactory {
+}
+
+#[derive(Debug)]
+struct LoadError {
```
I removed this in 87979ef65eb9eb3958f1ead564a663ba2bd117ad; any reason for it to return?

nox commented:
Absolutely no idea.

Cc @edunham @larsbergstrom.

Ms2ger commented:
@bors-servo r-

nox commented:
Ugh, I'll never use Reviewable again.

nox commented:
@Ms2ger Addressed your remarks.

Ms2ger commented:
Thanks, fine for me.

Manishearth commented:
@bors-servo treeclosed-

bors-servo commented:
:umbrella: The latest upstream changes (presumably #15844) made this pull request unmergeable. Please resolve the merge conflicts.

nox commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 23d19f2 with merge 2920253...

bors-servo commented:
:broken_heart: Test failed - [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1236)

nox commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit bccf5b3 with merge af81a9b...

bors-servo commented:
:broken_heart: Test failed - [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1238)

bors-servo commented:
:umbrella: The latest upstream changes (presumably #15920) made this pull request unmergeable. Please resolve the merge conflicts.

nox commented:
@metajack What's the status with your custom openssl archive?

nox commented:
@bors-servo r=nox,metajack,Ms2ger

bors-servo commented:
:pushpin: Commit 6c39100 has been approved by `nox,metajack,Ms2ger`

<!-- @bors-servo r=nox,metajack,Ms2ger 6c391003a88cf1c03afb56625634ef7921b82bbf -->

bors-servo commented:
:hourglass: Testing commit 6c39100 with merge 2bbbaa8...

bors-servo commented:
:broken_heart: Test failed - [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1310)

nox commented:
@bors-servo r=nox,metajack,Ms2ger

bors-servo commented:
:pushpin: Commit cd2a8f6 has been approved by `nox,metajack,Ms2ger`

<!-- @bors-servo r=nox,metajack,Ms2ger cd2a8f6038b3ba93039a8a3614534fce75d626a2 -->

bors-servo commented:
:hourglass: Testing commit cd2a8f6 with merge 7a4dad8...

bors-servo commented:
:broken_heart: Test failed - [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1313)

nox commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit b3c033f with merge 06cc212...

bors-servo commented:
:broken_heart: Test failed - [windows-gnu-dev](http://build.servo.org/builders/windows-gnu-dev/builds/1309)

bors-servo commented:
:umbrella: The latest upstream changes (presumably #15938) made this pull request unmergeable. Please resolve the merge conflicts.

avadacatavra commented:
This is now blocked on switching to ws (#14517) per discussion here: http://logs.glob.uno/?c=mozilla%23servo#c632460

avadacatavra commented:
This is now blocked on switching to ws (#14517) per discussion here: http://logs.glob.uno/?c=mozilla%23servo#c632460

nox committed:
Disable wss support

None of the existing websocket libraries are currently able to cope
with openssl 0.9 and be featureful, so we are disabling wss for now
to land openssl 0.9.

nox committed:
Revert "Store SSL context in ResourceGroup"

This reverts commit 02b2aa159a2019836e799c73ead43f2d189c9e2a.

nox committed:
Revert "Introduce create_ssl_context"

This reverts commit 7a4632bfa22c8bb83fbfe0c0241411362ee4dffd.

nox commented:
r? @jdm

nox commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit e527c9a991c00a3d01fcf262b9fee0b5c948436e with merge adee7e52989182a933a5f5ba848aeb4d03075f41...

jdm commented:
@bors-servo: r+

bors-servo commented:
:pushpin: Commit e527c9a has been approved by `jdm`

<!-- @bors-servo r=jdm e527c9a991c00a3d01fcf262b9fee0b5c948436e -->

nox commented:
@bors-servo try- retry

bors-servo commented:
:hourglass: Testing commit e527c9a991c00a3d01fcf262b9fee0b5c948436e with merge 5c3e5497af6c8a226b0ec8d48760e54c32af9d3d...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/3181)

nox commented:
@bors-servo retry #14323

bors-servo commented:
:hourglass: Testing commit e527c9a991c00a3d01fcf262b9fee0b5c948436e with merge 82b0d5ad54d94a29f595d59cbb37dcbab5d5a5c8...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/6075), [arm32](http://build.servo.org/builders/arm32/builds/6295), [arm64](http://build.servo.org/builders/arm64/builds/6328), [linux-dev](http://build.servo.org/builders/linux-dev/builds/6430), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/3161), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/3183), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/6434), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/6464), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/3120), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/3185), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1603)
Approved by: jdm
Pushing 82b0d5ad54d94a29f595d59cbb37dcbab5d5a5c8 to master...
